### PR TITLE
Add text debounce to QuickFilterView & ComboQuickFilterView

### PR DESCRIPTION
**Pull request by TristanCrawford**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

Using the quick filter in places like the Strings window can be painfully laggy with a large number of list items due to the fact that the list is updated on *every* keystroke.

I adapted the code [here](https://wiki.qt.io/Delay_action_to_wait_for_user_interaction) to debounce the input in both of these widgets.

I chose to use a hard-coded value of ~300ms based on the average typist's CPM of 200~ 150ms, but this could be turned into a setting if desired.

**Test plan (required)**

- Open Cutter
- Load a binary (preferably with a large amount of strings)
- Navigate to the Strings window (or any other window that uses a filter widget)
- Type in some text and observe that the filter behaves as expected, but without updating the list on every keystroke.

**Closing issues**

N/A (I couldn't find any relating to this)

### CODE REVIEW

File: src/widgets/ComboQuickFilterView.cpp [hunk 1]
```diff
@@ -6,8 +6,16 @@ ComboQuickFilterView::ComboQuickFilterView(QWidget *parent)
...
-    connect(ui->lineEdit, &QLineEdit::textChanged, this,
-            [this](const QString &text) { emit filterTextChanged(text); });
+    m_debounceTimer = new QTimer(this);
+    m_debounceTimer->setSingleShot(true);
+
+    connect(m_debounceTimer, &QTimer::timeout, this,
+            [this]() { emit filterTextChanged(m_filterText); });
+
+    connect(ui->lineEdit, &QLineEdit::textChanged, this, [this](const QString &text) {
+        m_filterText = text;
```
Comment (karliss) [inline src/widgets/ComboQuickFilterView.cpp]: No need to introduce additional field for saving text value. You might as well pass the latest value from `ui->lineEdit->text()` in the timeout handler.

File: src/widgets/ComboQuickFilterView.h [hunk 2]
```diff
@@ -32,6 +33,8 @@ public slots:
 
 private:
     Ui::ComboQuickFilterView *ui;
+    QTimer *m_debounceTimer;
+    QString m_filterText;
```
Comment (karliss) [inline src/widgets/ComboQuickFilterView.h]: For the most part Cutter coding style doesn't use "m_" prefix for class member.

**Review by karliss — COMMENTED** [review]
Other than the minor nitpicks seems fine.

300ms is long enough too be noticable that it's less responsive than before for views with short lists. But it's less bad than not being able to type in full word in the filters for big lists.

File: src/widgets/ComboQuickFilterView.cpp (hunk 1 above)
Comment (TristanCrawford) [reply src/widgets/ComboQuickFilterView.cpp]: Fixed.
https://github.com/rizinorg/cutter/blob/1edd3765396579c5cf9429e0602f9af4d638812a/src/widgets/QuickFilterView.cpp#L15-L16
https://github.com/rizinorg/cutter/blob/1edd3765396579c5cf9429e0602f9af4d638812a/src/widgets/ComboQuickFilterView.cpp#L12-L13

File: src/widgets/ComboQuickFilterView.h (hunk 2 above)
Comment (TristanCrawford) [reply src/widgets/ComboQuickFilterView.h]: Fixed.
https://github.com/rizinorg/cutter/blob/1edd3765396579c5cf9429e0602f9af4d638812a/src/widgets/QuickFilterView.h#L35
https://github.com/rizinorg/cutter/blob/1edd3765396579c5cf9429e0602f9af4d638812a/src/widgets/ComboQuickFilterView.h#L36

**Comment by TristanCrawford** [comment]
@karliss 
> Other than the minor nitpicks seems fine.
> 
> 300ms is long enough too be noticable that it's less responsive than before for views with short lists. But it's less bad than not being able to type in full word in the filters for big lists.

Cut it down to 150ms and it feels a bit more responsive now, lmk what you think.

Other feedback has also been addressed.

**Review by thestr4ng3r — APPROVED** [review]
Tested and 150ms seems like a good compromise to me. Thanks!

**Pull request merged**